### PR TITLE
Ruby 2.6 suport for 5-2-stable branch

### PR DESCRIPTION
**Pull request by yahonda**
### Summary

[Ruby 2.6.0-rc2 Released](https://www.ruby-lang.org/en/news/2018/12/15/ruby-2-6-0-rc2-released/) and Ruby 2.6 will be released on this December 25. I expect Rails 5.2 should run with Ruby 2.6 without errors/warnings.

This pull request backports changes to 5-2-stable branch to suppress errors and warnings which have been already addressed in the master branch. I'm wondering if these kinds of backporting should be done by Rails committers and/or release manager. Let me open a pull request first.

### Other Information

There are some errors remained. One can be "fixed" but it needs to backport pull requests with CHANGELOG. Another one, I do not know why it fails with 5-2-stable.

* It has been addressed by #32938 to master branch. It has changelog entry then I did not backport it to this pull request.

```ruby
$ cd activesupport
$ bundle exec ruby -w -Itest test/core_ext/range_ext_test.rb -n /test_should_compare/
Run options: -n /test_should_compare/ --seed 24989

# Running:

F

Failure:
RangeTest#test_should_compare_other_with_exclusive_end [test/core_ext/range_ext_test.rb:72]:
Expected false to be truthy.


bin/rails test test/core_ext/range_ext_test.rb:71

F

Failure:
RangeTest#test_should_compare_identical_exclusive [test/core_ext/range_ext_test.rb:68]:
Expected false to be truthy.


bin/rails test test/core_ext/range_ext_test.rb:67

F

Failure:
RangeTest#test_should_compare_identical_inclusive [test/core_ext/range_ext_test.rb:64]:
Expected false to be truthy.


bin/rails test test/core_ext/range_ext_test.rb:63



Finished in 0.001701s, 1763.5357 runs/s, 1763.5357 assertions/s.
3 runs, 3 assertions, 3 failures, 0 errors, 0 skips
```

* `test_restart_rails_server_with_custom_pid_file_path` fails 
It passes with the master branch. 

```ruby
$ cd railties
$ bundle exec ruby -w -Itest test/application/server_test.rb -n test_restart_rails_server_with_custom_pid_file_path
Run options: -n test_restart_rails_server_with_custom_pid_file_path --seed 41272

# Running:

F

Failure:
ApplicationTests::ServerTest#test_restart_rails_server_with_custom_pid_file_path [test/application/server_test.rb:53]:
"Inherited" expected, but got:

...
Traceback (most recent call last):
	2: from bin/rails:4:in `<main>'
	1: from /home/yahonda/.rbenv/versions/2.6.0-rc2/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/home/yahonda/.rbenv/versions/2.6.0-rc2/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- rails/commands (LoadError)
.
Expected "...\r\n\e[1mTraceback\e[m (most recent call last):\r\n\t2: from bin/rails:4:in `<main>'\r\n\t1: from /home/yahonda/.rbenv/versions/2.6.0-rc2/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'\r\n/home/yahonda/.rbenv/versions/2.6.0-rc2/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require': \e[1mcannot load such file -- rails/commands (\e[1;4mLoadError\e[m\e[1m)\e[m\r\n" to include "Inherited".


bin/rails test test/application/server_test.rb:35



Finished in 14.595938s, 0.0685 runs/s, 0.4111 assertions/s.
1 runs, 6 assertions, 1 failures, 0 errors, 0 skips
```

### CODE REVIEW

**Comment by yahonda** [comment]
Looks like this pull request includes something not available for Ruby 2.2 and 2.3. Let me correct them.

**Comment by yahonda** [comment]
Reverted https://github.com/rails/rails/pull/34720/commits/e8237d18f3f756e49d4db91e4a69bc9166f604a5 since only Ruby 2.4 or higher supports `Regexp#match?`. Now the remaining failures at 5-2-stable are #32938 and #34719.

**Comment by ahorek** [comment]
Regexp#match? is supported https://github.com/rails/rails/blob/5-2-stable/activesupport/lib/active_support/core_ext/regexp.rb#L8 but String#match? isn't

I don't see a reason why https://github.com/rails/rails/pull/32695 should be backported. It's a performance optimization and warnings are from rubocop. It's not related to Ruby 2.6 compatibility at all.

**Comment by yahonda** [comment]
Thanks for the reviews.

I have dropped #32695, ff0af71 and 48d1c48 from this pull request changing .rubocop.yml
Also backported #32938 to 5-2-stable to support Ruby 2.6 new Range behavior.

File: activesupport/lib/active_support/core_ext/range/include_range.rb [hunk 1]
```diff
@@ -1,25 +1,9 @@
...
-      else
-        super
-      end
-    end
-  end
-end
+require "active_support/deprecation"
 
-Range.prepend(ActiveSupport::IncludeWithRange)
+ActiveSupport::Deprecation.warn "You have required `active_support/core_ext/range/include_range`. " \
```
Comment (kamipo) [inline activesupport/lib/active_support/core_ext/range/include_range.rb]: The deprecation warning should not be happened in the next stable release.

Comment (yahonda) [reply activesupport/lib/active_support/core_ext/range/include_range.rb]: Removed this deprecation warning as far as I understand this file will not be removed in 5.2.x.

Comment (y-yagi) [reply activesupport/lib/active_support/core_ext/range/include_range.rb]: Can you fix CHANGELOG entry too?
https://github.com/rails/rails/pull/34720/files#diff-4be3c90df95ede75a8d48016f370c0d9R7

Comment (yahonda) [reply activesupport/lib/active_support/core_ext/range/include_range.rb]: Updated the changelog entry.

**Pull request merged**